core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -141,7 +141,8 @@ class GroupMetadataManager(val brokerId: Int,
       // if we crash or leaders move) since the new leaders will still expire the consumers with heartbeat and
       // retry removing this group.
       val groupPartition = partitionFor(group.groupId)
-      val tombstone = new Message(bytes = null, key = GroupMetadataManager.groupMetadataKey(group.groupId))
+      val tombstone = new Message(bytes = null, key = GroupMetadataManager.groupMetadataKey(group.groupId),
+        timestamp = SystemTime.milliseconds, magicValue = Message.MagicValue_V1)
```

Comment:
We should use the message format specified for the internal topic. In addition, we have to be a bit careful here, during a rolling upgrade, if a broker's inter-broker protocol version is still before 0.10.0, even if the message format is for v1, we will still want to use message format v0. Otherwise, if another broker is still on the pre 0.10.0 code, it can't read the v1 message in the internal topic. Ditto to the magic value setting below.